### PR TITLE
Don't update pins and elements if changes are outside viewed area

### PR DESCRIPTION
**Pull request by Helium314**
This is the "straightforward useful" part of #4283, plus (more or less) the same thing for `StyleableOverlayManager`, as per https://github.com/streetcomplete/StreetComplete/pull/4283#pullrequestreview-1071963887.

I did *not yet* test this, other than whether it compiles.

### CODE REVIEW

File: app/src/main/java/de/westnordost/streetcomplete/screens/main/map/QuestPinsManager.kt [hunk 1]
```diff
@@ -140,11 +141,16 @@ class QuestPinsManager(
...
         val pins = synchronized(questsInView) {
-            added.forEach { questsInView[it.key] = createQuestPins(it) }
-            removed.forEach { questsInView.remove(it) }
+            addedInView.forEach { questsInView[it.key] = createQuestPins(it) }
+            removed.forEach { if (questsInView.remove(it) != null) deletedAny = true }
             questsInView.values.flatten()
         }
+        if (!deletedAny && addedInView.isEmpty())
+            return
         pinsMapComponent.set(pins)
```
Comment (westnordost) [inline app/src/main/java/de/westnordost/streetcomplete/screens/main/map/QuestPinsManager.kt:154]:
```suggestion
        if (deletedAny || addedInView.isNotEmpty())
            pinsMapComponent.set(pins)
```

Just to make it consistent with the code in StyleableOverlayManager

**Review by westnordost — APPROVED** [review]
Code looks good! Of course, as always, it must be tested before it is merged.

**Comment by Helium314** [comment]
Not sure if this is enough: I just used SC with this PR, and didn't encounter any issues when uploading (triggers the call to `updateQuestPins` / `updateStyledElements`). Done with and without overlay enabled.

**Comment by westnordost** [comment]
Hmm, i think this is enough,  than you!

**Pull request merged**